dashboard/modules/reporter/reporter_agent.py

```diff
@@ -678,24 +678,29 @@ def _record_stats(self, stats, cluster_stats):
                 network_receive_speed_record,
             ]
         )
-        return records_reported
+        return records_to_report
 
     async def _perform_iteration(self, publisher):
         """Get any changes to the log files and push updates to kv."""
         while True:
             try:
-                formatted_status_string = internal_kv._internal_kv_get(
-                    DEBUG_AUTOSCALING_STATUS
-                )
-                cluster_stats = (
-                    json.loads(formatted_status_string.decode())
-                    if formatted_status_string
-                    else {}
-                )
+                # Cluster stats don't need to be recorded on all agents,
+                # so we only report it in the head node.
+                if self._is_head_node:
+                    formatted_status_string = internal_kv._internal_kv_get(
+                        DEBUG_AUTOSCALING_STATUS
+                    )
+                    cluster_stats = (
+                        json.loads(formatted_status_string.decode())
+                        if formatted_status_string
+                        else {}
+                    )
+                else:
+                    cluster_stats = {}
 
                 stats = self._get_all_stats()
-                records_reported = self._record_stats(stats, cluster_stats)
-                self._metrics_agent.record_reporter_stats(records_reported)
+                records_to_report = self._get_records_to_report(stats, cluster_stats)
```

Comment:
In this case, what about we just split `_get_records_to_report ` into two functions? 

```python3
if head:
    cluster_stats_to_report = self._get_records_to_report(stats, cluster_stats)
    self._metrics_agent.record_reporter_stats(cluster_stats_to_report)
node_stats_to_report = self._get_records_to_report(stats, cluster_stats)
self._metrics_agent.record_reporter_stats(node_stats_to_report)
```

Reply:
Maybe I should just change cluster_stats to an optional parameter then